### DISCUSSION TITLE
Irradiation level

### DISCUSSION THREAD

**Discussion by darek-margas** [BJReplay/ha-solcast-solar#134] [Ideas]
I wonder it it would be possible to have irradiation - either day total, growing as it goes or in time slots. It would be useful for other things such as automated irrigation.

**Comment by gcoan** [comment]
> I wonder it it would be possible to have irradiation - either day total, growing as it goes or in time slots. It would be useful for other things such as automated irrigation.  

Solcast would only be able to give forecast solar generation, and you could template up a sum so far today helper if you needed it. 
Of course it's only a forecast, if you have solar panels the actual from your inverter is usually more accurate

**Comment by BJReplay** [comment]
Solcast is a great service for forecasting power output from Solar Panels.  Yes, it does take into account the likelihood of cloud, so I can see how it would be useful for irrigation, but where it fails there is that a proper solcast setup has an azimuth and elevation for panels that simply don't apply for irrigation purposes.

Finally, it is a forecast rather than actuals.

Weather stations will often report irradiation (and HA integrations exist for them, for example for Ecowitt) and integrating the area under the chart for this is a better indication (when combined with rainfall, humidity, wind) as indicators of need for irrigation).

Possible? Yes.
Desirable? I don't think so.

![image](https://github.com/user-attachments/assets/0bc8ddbb-4e93-408a-bb8e-3fba22785898)

**Reply by darek-margas** [reply]
Well, forecast is for the future but there is data for the past, Not much different from the above. As values are needed at the end of the day, to calculate evaporation which has happened over last day, it could be same good as local station. Reading inverter is a bit problematic if its capacity is below panels capacity, it won't be possible to establish irradiation when the inverter hits the limit.
![image](https://github.com/user-attachments/assets/30944fab-2abf-4ff4-93b1-2a80ee41b904)

**Reply by darek-margas** [reply]
Pretty much this: https://kb.solcast.com.au/the-difference-between-estimated-actuals-and-forecasts
As they explain query for previous day will give it but late evening same day should do too.

**Reply by darek-margas** [reply]
![image](https://github.com/user-attachments/assets/8197113d-e00c-4007-a3c4-ced7ca71e70f)

**Reply by BJReplay** [reply]
Not available for hobbyists.

Will not be supported.
